### ISSUE TITLE
Have to enter 2 times to execute command on startup

### ISSUE THREAD

**Issue by HowardDo2208** [warpdotdev/Warp#1694]
### Describe the bug

On app startup or opening a new tab, every command that I type in isn't syntax highlighted when I first press Enter doesn't execute right away but it minimizes the lines a little, and only when I press enter 1 more time then it run.

The problem would be resolved if I restart the shell with `exec zsh`, but it is annoying to do this every time opening up Warp

### To Reproduce

1. Open warp
2. Type in any command
3. Press enter
4. Have to press enter again to execute it

### Expected behaviour

It should execute right away, and come with syntax highlighting

### Screenshots

![image](https://user-images.githubusercontent.com/62332195/183352299-33304376-2473-49bf-be1e-114751b6118c.png)
after the first enter pressed:
![image](https://user-images.githubusercontent.com/62332195/183352365-f6a21cb0-7b58-4005-a25e-61b465c5f33d.png)
second enter:
![image](https://user-images.githubusercontent.com/62332195/183352408-e75ec377-782b-4d07-9a11-d60fb4195634.png)

### Operating System

MacOS

### OS Version

12.5

### Shell Version

zsh 5.8.1 (x86_64-apple-darwin21.0)

### Warp Version

v0.2022.08.01.09.12.stable_02

### Does this block you from using Warp daily?

Yes

**Comment by zheng** [comment]
Sorry for this issue! My guess is that there's something in your zsh configs we're not handling well. Does this still happen if you start with no configs? You can do this quickly by putting editing `~/.zshenv` to set `ZDOTDIR=/`

**Comment by HowardDo2208** [comment]
> Sorry for this issue! My guess is that there's something in your zsh configs we're not handling well. Does this still happen if you start with no configs? You can do this quickly by putting editing `~/.zshenv` to set `ZDOTDIR=/`

Yeah, I did put my zsh config files in a separate folder (~/.config/zsh), after using the default path the problem is resolved, but I would like to keep my old config.

**Comment by zheng** [comment]
Do you know what in the config causes the issue? We’d love to fix the issue
in Warp, but need a repro first

**Comment by HowardDo2208** [comment]
Here is my zsh config: https://github.com/HowardDo2208/zshconf
I set it up  from a youtube tutorial: https://youtu.be/bTLYiNvRIVI

**Comment by warpdotdev-dx** [comment]
Can you help us drill down on what's causing this?


The easiest way to do this is to move your config settings behind this terminal based conditional flag:
if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
##### WHAT YOU WANT TO DISABLE FOR WARP - BELOW


# Whatever you want to disable


##### WHAT YOU WANT TO DISABLE FOR WARP - ABOVE
fi




After doing so you can move things out of the flag so that they are enabled for Warp until we find what's breaking it.


See our documentation here: 
<https://docs.warp.dev/help/known-issues#debugging>

**Comment by vieitesss** [comment]
Hi!

I have the same issue. After searching the lines of the .zshrc with the conditional flag that @warpdotdev-dx provided, I found that these lines are the problem.

```
source "$(brew --prefix)"/opt/zsh-vi-mode/share/zsh-vi-mode/zsh-vi-mode.plugin.zsh
zvm_bindkey viins '^U' end-of-line
```

I use this plugin to modify the command that I'm writing using vim motions.

**Comment by e-cal** [comment]
shoutout @jamapag, looks like warp just doesn't handle and ^j (ctrl+j) binds well

**Comment by cloud303-cholden** [comment]
> I had a same issue, fixed by removing this binding:
> 
> `bindkey '^J' self-insert`

Removing `bindkey -r "^J"` fixed it for me. Thanks!

**Comment by Hirad2hhk** [comment]
as of today the issue is still there. the problem is with key bindings. when I set bindings in my zshrc on mac every command takes two enters to run and when I remove them its fine again

**Comment by omerxx** [comment]
same here. zshrc gone but warp requires 2 enter hits to run a command 😵‍💫

**Comment by bahag-chaurasiak** [comment]
@omerxx commenting on your .zshrc (#bindkey '^j' down-line-or-search) should work, then after simple restart the warp.
